sound.js

```diff
@@ -230,6 +243,30 @@ Sound.registerHeadsetPlugChangeListener = function(headsetPluggedInListener) {
   }
 };
 
+Sound.prototype.registerOnPlayListener = function(onPlayListener) {
+  if (this.onPlaySubscription != null) {
+    console.warn('On Play change event Listener is already registered');
+    return;
+  }
+
+  if(!IsWindows) {
```

Comment:
You only added event listener for Android, so this code should have been `if(IsAndroid)`. This might not work on iOS.